packages/js/email-editor/src/components/sidebar/details-panel.tsx

```diff
@@ -35,6 +35,220 @@ export function DetailsPanel() {
 
 	const { togglePersonalizationTagsModal, updateEmailMailPoetProperty } =
 		useDispatch( storeName );
+	const [ activeRichText, setActiveRichText ] = useState( null );
+	const [ selectionRange, setSelectionRange ] = useState( null );
+
+	const subjectRef = useRef( null );
+	const preheaderRef = useRef( null );
+
+	// Convert `RichText` DOM offset to stored value offset
+	const mapRichTextToValue = ( html ) => {
```

Comment:
I would suggest using `React.useCallback` for the long functions defined within a functional component. Without the useCallback the function is recreated on every re-render of the component.